### ISSUE TITLE
video has image but no sound when play hls

### ISSUE THREAD

**Issue by lanhe233** [videojs/video.js#6090]
## Description
I use videojs to play hls, but when I click play button, video has no sound. And video has no image until five seconds later. After 5 seconds, video plays normal, and also has no sound.

## Steps to reproduce
1. use videojs() to initialize
2. player.src({
   src: 'xxxxxxx.m3u8‘,
   type: 'application/x-mpegURL'
})
3. I tried some versions > 7.0.0

### Expected
I want video can play normal and have sound

### Actual
lose image during 0-5 seconds, and no sound during whole duration

### Error output
This error repeat twelve times after I click play button, and during this time, video has no image.
VIDEOJS: ERROR: TypeError: Cannot read property 'attributes' of null
    at SegmentLoader.<anonymous> (video7.6.0.js:58493)
    at SegmentLoader.data.dispatcher (video7.6.0.js:1824)
    at trigger (video7.6.0.js:1960)
    at SegmentLoader.EventTarget.trigger (video7.6.0.js:2479)
    at SegmentLoader.handleUpdateEnd_ (video7.6.0.js:55481)
    at video7.6.0.js:53096
    at VirtualSourceBuffer.onUpdateendCallback_ (video7.6.0.js:53044)
    at VirtualSourceBuffer.data.dispatcher (video7.6.0.js:1824)
    at trigger (video7.6.0.js:1960)
    at VirtualSourceBuffer.EventTarget.trigger (video7.6.0.js:2479)

## Additional Information
_playlist.m3u8_
#EXTM3U
#EXT-X-VERSION:3

#EXT-X-STREAM-INF:BANDWIDTH=2000000,CODECS="avc1.4D401F,",RESOLUTION=1280x720
playlist_origin.stream.m3u8

_playlist_origin.stream.m3u8_
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-ALLOW-CACHE:NO
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-TARGETDURATION:5

#EXTINF:0.50099998712539673,
origin.stream_0.ts
#EXTINF:4.8380002975463867,
origin.stream_1.ts
#EXTINF:3.6700000762939453,
origin.stream_2.ts
#EXTINF:1.6679999828338623,
origin.stream_3.ts
#EXTINF:4.0460000038146973,
origin.stream_4.ts
#EXTINF:1.7519999742507935,
origin.stream_5.ts
#EXTINF:3.9200000762939453,
origin.stream_6.ts
#EXTINF:0.95999997854232788,
origin.stream_7.ts
#EXTINF:2.127000093460083,
origin.stream_8.ts
#EXTINF:1.1679999828338623,
origin.stream_9.ts
#EXTINF:2.0429999828338623,
origin.stream_10.ts
#EXTINF:1.7940000295639038,
origin.stream_11.ts
#EXTINF:1.5430001020431519,
origin.stream_12.ts
#EXT-X-ENDLIST

#### videojs
@7.6.0
#### browsers
Chrome 75.0.3770.100
#### OSes
windows10
### plugins
no plugins

**Comment by lanhe233** [comment]
Today I tried firefox and edge. 
Firefox plays with no sound, but image is normal.
Edge plays normal including sound and image!

**Comment by gkatsev** [comment]
We just made a fix that had a similar behavior to what you're describing (https://github.com/videojs/mux.js/pull/285) but it's hard to know for sure without a test stream.

**Comment by lanhe233** [comment]
@gkatsev 
I create a simple test demo in : https://lanhe233.github.io/.
I hope it's helpful.
Thanks.

And, for my project, I find a temporary solution.
Import [videojs-contrib-hlsjs](https://github.com/Peer5/videojs-contrib-hls.js) after videojs...

**Comment by gkatsev** [comment]
That's super helpful, thanks! I can see that using the media playlist directly works. I see the error that you posted and it seems to do with rendition selection failing with just the one media playlist or something. We'll take a look.

**Comment by ProMasoud** [comment]
same here any fix?

**Comment by gkatsev** [comment]
Looked at it again and there are a couple of issues here.

1. null check
```
 VIDEOJS: ERROR: TypeError: Cannot read property 'attributes' of null
```
is a missing null check that I have a local fix for.

2. The missing audio
This is because in the master manifest there isn't an audio codec provided only a video codec. If I add an audio codec, it plays back fine.
From
```
#EXT-X-STREAM-INF:BANDWIDTH=2000000,CODECS="avc1.4D401F,",RESOLUTION=1280x720
```
to
```
#EXT-X-STREAM-INF:BANDWIDTH=2000000,CODECS="avc1.4D401F,mp4a.40.2",RESOLUTION=1280x720
```

3. there's a gap happening at around 23 seconds

4. the stream starts with 5 seconds of black during initial playback but it fine on replays.

**Comment by gkatsev** [comment]
Additionally, the example stream provided has a lot of issues that make it hard to play. For example, a bunch of the segments do not start with an iframe. It's highly recommended that each one does.
In addition to the fix we're making in videojs/http-streaming#779, both video and audio codecs should be provided in the master manifest.

Apple's media stream validator also complains about the mismatch between the peak and average bandwidth and the specified bandwidth.

Unfortunately, given the inherent issues in the stream, we are unlikely to be able to spend more time fixing low level issues. I would advise to re-transcode the video following HLS best practices (like iframes on every segment).
